addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -830,15 +830,15 @@ export function getDeepestPosition(node, offset) {
     let direction = DIRECTIONS.RIGHT;
     let next = node;
     while (next) {
-        if (isVisible(next) || isZWS(next)) {
+        if ((isVisible(next) || isZWS(next)) && (!isBlock(next) || next.isContentEditable)) {
             // Valid node: update position then try to go deeper.
             if (next !== node) {
                 [node, offset] = [next, direction ? 0 : nodeSize(next)];
             }
             // First switch direction to left if offset is at the end.
             direction = offset < node.childNodes.length;
             next = node.childNodes[direction ? offset : offset - 1];
-        } else if (direction && next.nextSibling && !isBlock(next.nextSibling)) {
+        } else if (direction && next.nextSibling && (!isBlock(next.nextSibling) || !next.isContentEditable)) {
```

Comment:
Semantics and style so feel free not to listen to me, but I prefer `!(isBlock(next.nextSibling) && next.isContentEditable)`. Most of the time I prefer `!(a && b)` over `!a || !b` because it's more intuitive: what we want is to not have the situation where these two things are true at the same time. It's harder to say it the other way. But that's me.

Reply:
I have removed these specific changes, so there is no longer a need to modify it.